src/materials/ConcreteExpansionMicrocrackingDamage.C

```diff
@@ -0,0 +1,167 @@
+/****************************************************************/
+/*               DO NOT MODIFY THIS HEADER                      */
+/*                       BlackBear                              */
+/*                                                              */
+/*           (c) 2017 Battelle Energy Alliance, LLC             */
+/*                   ALL RIGHTS RESERVED                        */
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+
+#include "ConcreteExpansionMicrocrackingDamage.h"
+#include "ElasticityTensorTools.h"
+
+registerMooseObject("BlackBearApp", ConcreteExpansionMicrocrackingDamage);
+
+template <>
+InputParameters
+validParams<ConcreteExpansionMicrocrackingDamage>()
+{
+  InputParameters params = validParams<ScalarDamageBase>();
+  params.addClassDescription("Scalar damage model based on extent of internal expansion");
+
+  params.addRequiredParam<MaterialPropertyName>(
+      "eigenstrain_name", "name of the eigenstrain driving the microcracking damage process");
+
+  params.addParam<bool>("use_isotropic_expansion",
+                        true,
+                        "indicates whether the model assumes an isotropic expansion (true) or "
+                        "decomposes the expansion alongside the principal direction (false)");
+
+  params.addParam<bool>("use_stress_control",
+                        true,
+                        "indicates whether the damage is affected by the current stress state");
+
+  params.addParam<Real>("correction_factor",
+                        1.0,
+                        "correction factor by which the eigenstrain is multiplied before "
+                        "evaluating the damage");
+
+  params.addRequiredRangeCheckedParam<Real>(
+      "microcracking_strain_init",
+      "microcracking_strain_init > 0",
+      "linear strain at which the microcracking initiates (in [m/m])");
+
+  params.addRequiredRangeCheckedParam<Real>(
+      "microcracking_strain_branch",
```

Comment:
It's not clear to me where the name 'branch' comes from in this parameter.  We should probably either rename this parameter, or somehow explain that.

Reply:
This parameter controls the slope of the damage as a function of the expansion. It has the same dimension as a strain. I wasn't sure how to name it either...